src/AudioBuffer.js

```diff
@@ -1,31 +1,50 @@
-var _ = require('underscore')
-
 class AudioBuffer {
+  #sampleRate
+  get sampleRate() { return this.#sampleRate }
+  #length
+  get length() { return this.#length }
+  #duration
+  get duration() { return this.#duration }
+  #numberOfChannels
+  get numberOfChannels() { return this.#numberOfChannels }
 
   constructor(numberOfChannels, length, sampleRate) {
     var ch
-    this._data = []
+
     // Just a hack to be able to create a partially initialized AudioBuffer
-    if (arguments.length) {
-      for (ch = 0; ch < numberOfChannels; ch++)
-        this._data.push(new Float32Array(length))
-      this._defineAttrs(numberOfChannels, length, sampleRate)
-    }
+    if (!arguments.length) return
+
+    // define attrs
+    if (!(sampleRate > 0)) throw new Error('invalid sample rate : ' + sampleRate)
+    this.#sampleRate = sampleRate
+    if (!(length >= 0)) throw new Error('invalid length : ' + length)
+    this.#length = length
+    this.#duration = length / sampleRate
+    if (!(numberOfChannels > 0)) throw new Error('invalid numberOfChannels : ' + numberOfChannels)
+    this.#numberOfChannels = numberOfChannels
+
+    //data is stored as a planar sequence
+    this._buffer = new Float32Array(this.length * this.numberOfChannels)
```

Comment:
Why `_buffer` and not `#buffer` ?

Reply:
That may be needed for DSP processing, we may need to pass single array as chunk of memory.

Reply:
Thanks for the explanation 👍